File: 프로그래머스/3/43164. 여행경로/여행경로.py
def solution(tickets):
    answer = []
    N = len(tickets)
    visited = [False for _ in range(N)]
    def dfs(curr, route, dept):
        ## print(route, dept)
        if(dept==N):
            answer.append(" ".join(route))
            
        for idx,next_ in enumerate(tickets):
            [start, end]=next_
            
            if(curr!=start or visited[idx]):
                continue
            
            visited[idx]=True
            dfs(end, route+[end], dept+1)
            visited[idx]=False
    dfs("ICN",["ICN"],0)
    answer.sort()
    ### print(answer)
    return answer[0].split(" ")

File: 프로그래머스/3/43164. 여행경로/여행경로.js
function solution(tickets) {
    var answer = [];
    const N = tickets.length
    const visited = Array.from({length:N}).fill(false)
    const routes = []
    function dfs(curr, route, dept){
        if(dept==N){
            answer.push(route.join(" "))
            return
        }
        for(let idx=0;idx<N;idx++){
            const next_ =  tickets[idx]
            if(visited[idx] || next_[0]!=curr)continue
            visited[idx] = true
            dfs(next_[1], [...route,next_[1]], dept+1 )
            visited[idx]=false
        }
    }
    dfs("ICN",["ICN"],0)
    answer.sort()
    return answer[0].split(" ");
}